reana_client/validation.py

```diff
@@ -343,6 +344,33 @@ def _warn_not_defined_parameters(
         )
 
 
+def _validate_cwl_parameters(reana_yaml):
+    """Validate input parameters for CWL workflows.
+
+    :param reana_yaml: REANA YAML specification.
+    """
+
+    cwl_files = set()
+    cwl_main_spec_path = reana_yaml["workflow"]["file"]
+    cwl_graph = reana_yaml["workflow"]["specification"].get("$graph", None)
+
+    if cwl_graph:
+        cwl_directory = cwl_main_spec_path.rsplit("/", 1)[0]
+        for spec in cwl_graph:
+            spec_file = spec.get("id")[1:]
+            spec_file_path = cwl_directory + "/" + spec_file
+            cwl_files.add(spec_file_path)
```

Comment:
As we discussed yesterday, we don't need to check the files that are referenced inside the main workflow spec file as `cwltool` would take care of it for us. 

For instance, choosing `r-d-helloworld` as reference. If you modify `helloworld.tool`:
```diff
diff --git a/workflow/cwl/helloworld.tool b/workflow/cwl/helloworld.tool
index 4aa963c..60a8b6b 100644
--- a/workflow/cwl/helloworld.tool
+++ b/workflow/cwl/helloworld.tool
@@ -19,7 +19,7 @@ inputs:
     type: File
     inputBinding:
       prefix: --inputfile
-  sleeptime:
+  sleeptim:
     type: int
     inputBinding:
       prefix: --sleeptime
```

And the you run `cwltool` pointing **to the main workflow file**, it would already warn us that something is wrong:

```console
$ cwltool --validate --strict workflow/cwl/helloworld.cwl
INFO /Users/marco/.virtualenvs/r/bin/cwltool 1.0.20191022103248
INFO Resolved 'workflow/cwl/helloworld.cwl' to 'file:///Users/marco/code/reanahub/reana-demo-helloworld/workflow/cwl/helloworld.cwl'
ERROR Tool definition failed validation:
workflow/cwl/helloworld.cwl:34:5: Step is missing required parameter 'sleeptim'
```

So we don't need to iterate ourselves, it would be as simple as calling `cwltool` once with `workflow.file`.

Reply:
Yes, Adding the test rn. Thanks for reminding 👍